### PR TITLE
Added functionality to filter restricted courses and displaying on the page.

### PR DESCRIPTION
**Pull request by ashwinspencer**

### CODE REVIEW

File: client/src/components/Questions/Tabs/Checkbox.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import React, { Component, PropTypes } from 'react';
+
+class Checkbox extends Component {
```
Comment (dstjohniii) [inline client/src/components/Questions/Tabs/Checkbox.js:3]: Why aren't you using mui/material/Checkbox? creating your own will make styling harder. Also we shouldn't be using classes as its not good practice anymore.

File: client/src/components/Questions/Tabs/RestrictedCourses.js [hunk 2]
```diff
@@ -3,8 +3,45 @@ import Container from "@mui/material/Container";
 import FormGroup from "@mui/material/FormGroup";
 import FormControlLabel from "@mui/material/FormControlLabel";
 import Checkbox from "@mui/material/Checkbox";
+import parseCSVFile from '../../../csvParser';
+import dataFile from '../../../data/rotation_cleaned.csv';
```
Comment (dstjohniii) [inline client/src/components/Questions/Tabs/RestrictedCourses.js:7]: I dont believe these are being

File: client/src/components/Questions/Tabs/RestrictedCourses.js [hunk 3]
```diff
@@ -3,8 +3,45 @@ import Container from "@mui/material/Container";
...
+import courses from "../../../data/ClassInfo.json";
+
+function isRestricted(value){
+  let retValue=value.restricted
+  if(retValue=='true'){
+   retValue='true'
+  }
+  return retValue
+}
+
```
Comment (gwallace04) [inline client/src/components/Questions/Tabs/RestrictedCourses.js:17]: This is unnecessary. You can just return `value.restricted`

File: client/src/components/Questions/Tabs/Checkbox.js (hunk 1 above)
Comment (ashwinspencer) [reply client/src/components/Questions/Tabs/Checkbox.js:3]: I am using mui/material/Checkbox.
I don't see Classes anywhere in my code.

File: client/src/components/Questions/Tabs/RestrictedCourses.js (hunk 2 above)
Comment (ashwinspencer) [reply client/src/components/Questions/Tabs/RestrictedCourses.js]: Removed unused imports.

File: client/src/components/Questions/Tabs/RestrictedCourses.js (hunk 3 above)
Comment (ashwinspencer) [reply client/src/components/Questions/Tabs/RestrictedCourses.js:12]: Done.

**Review by dstjohniii — APPROVED** [review]
LGTM

**Pull request merged**